src/EnergyPlus/FanCoilUnits.cc

```diff
@@ -1938,13 +1941,14 @@ namespace FanCoilUnits {
 			}
 			// obtain unit output with no active heating/cooling
 			Calc4PipeFanCoil( FanCoilNum, ControlledZoneNum, FirstHVACIteration, QUnitOutNoHC, 0.0 );
-			if ( FirstHVACIteration ) {
-				FanCoil( FanCoilNum ).QUnitOutNoHC = QUnitOutNoHC;
-			}
-			else {
+
+			if ( ColdFlowLocked || HotFlowLocked ) {
 				QUnitOutNoHC = FanCoil( FanCoilNum ).QUnitOutNoHC;
+			} else {  // continue to update QUnitOutNoHC while flow is unlocked
+				FanCoil( FanCoilNum ).QUnitOutNoHC = QUnitOutNoHC;
```

Comment:
I think this may also impact other water coil models and there is no way to tell unless a user points something out. So to test I have been thinking about a fatal error in SetComponentFlowRate to see if there are failures in the example files (just to get us started on figuring this out). The problem is that when flow is locked, the water outlet node mass flow rate is set equal to the inlet node mass flow rate instead of some other value which we could compare against.

    } else if ( loop_side.FlowLock == FlowLocked ) {
        Node( OutletNode ).MassFlowRate = Node( InletNode ).MassFlowRate;
        CompFlow = Node( OutletNode ).MassFlowRate;
    } else {
        ShowFatalError( "SetComponentFlowRate: Flow lock out of range" ); //DEBUG error...should never get here
    }

when flow is locked down, if the Node() data held the value of the locked flow then we could compare inlet node flow to the locked value.

    } else if ( loop_side.FlowLock == FlowLocked ) {
        Node( OutletNode ).MassFlowRate = Node( InletNode ).LockedMassFlowRate;
        if(  Node( InletNode ).MassFlowRate !=  Node( InletNode ).LockedMassFlowRate ) ShowFatalError....

@EnergyArchmage and @mjwitte might have other suggestions.